### PR TITLE
[Federation]Fix e2e_federation: pass clusterIngress out

### PR DESCRIPTION
**Pull request by FengyunPan**
None

### CODE REVIEW

File: test/e2e_federation/ingress.go [hunk 1]
```diff
@@ -323,16 +323,17 @@ func waitForIngressOrFail(clientset *kubeclientset.Clientset, namespace string,
 	By(fmt.Sprintf("Fetching a federated ingress shard of ingress %q in namespace %q from cluster", ingress.Name, namespace))
 	var clusterIngress *v1beta1.Ingress
 	err := wait.PollImmediate(framework.Poll, timeout, func() (bool, error) {
-		clusterIngress, err := clientset.Ingresses(namespace).Get(ingress.Name, metav1.GetOptions{})
-		if (!present) && errors.IsNotFound(err) { // We want it gone, and it's gone.
+		var get_err error
+		clusterIngress, get_err = clientset.Ingresses(namespace).Get(ingress.Name, metav1.GetOptions{})
```
Comment (shashidharatd) [inline test/e2e_federation/ingress.go]: I think there was no need to change the `err` variable name to get_err. please revert it.

Comment (FengyunPan) [inline test/e2e_federation/ingress.go]: Alright.

File: test/e2e_federation/ingress.go [hunk 2]
```diff
@@ -322,8 +322,9 @@ func verifyCascadingDeletionForIngress(clientset *fedclientset.Clientset, cluste
 func waitForIngressOrFail(clientset *kubeclientset.Clientset, namespace string, ingress *v1beta1.Ingress, present bool, timeout time.Duration) {
 	By(fmt.Sprintf("Fetching a federated ingress shard of ingress %q in namespace %q from cluster", ingress.Name, namespace))
 	var clusterIngress *v1beta1.Ingress
-	err := wait.PollImmediate(framework.Poll, timeout, func() (bool, error) {
-		clusterIngress, err := clientset.Ingresses(namespace).Get(ingress.Name, metav1.GetOptions{})
+	var err error
+	err = wait.PollImmediate(framework.Poll, timeout, func() (bool, error) {
+		clusterIngress, err = clientset.Ingresses(namespace).Get(ingress.Name, metav1.GetOptions{})
```
Comment (shashidharatd) [inline test/e2e_federation/ingress.go]: Just put the `var err error` inside the func() closure. even though err is same name, they could be 2 different variables.

Comment (FengyunPan) [inline test/e2e_federation/ingress.go]: Done

**Comment by shashidharatd** [comment]
@k8s-bot ok to test

**Comment by shashidharatd** [comment]
/lgtm
Thanks @FengyunPan !

**Comment by FengyunPan** [comment]
@k8s-bot unit test this

**Comment by madhusudancs** [comment]
/lgtm cancel

Federation presubmit failed: https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/45005/pull-kubernetes-federation-e2e-gce/2790/

**Comment by madhusudancs** [comment]
@ks8-bot pull-kubernetes-federation-e2e-gce test this

**Comment by madhusudancs** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by FengyunPan** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Pull request merged**